conn.go

```diff
@@ -262,33 +357,125 @@ func connect(config ConnConfig, connInfo *pgtype.ConnInfo) (c *Conn, err error)
 		}
 	}
 
+	if c.config.TargetSessionAttrs != "" &&
```

Comment:
Defined `TargetSessionAttrs` property as custom type (`TargetSessionType`) instead of `string`. Now all validation logick is inside `isValid` method for this type.

I'm not sure, if I can do it better.